src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs

```diff
@@ -1557,7 +1557,14 @@ private static async Task InsertText(string code, string text, bool expectDocume
             Assert.Equal(expectDocumentAnalysis ? 1 : 0, analyzer.DocumentIds.Count);
         }
 
-        private static async Task<Analyzer> ExecuteOperation(TestWorkspace workspace, Action<TestWorkspace> operation)
+        private static Task<Analyzer> ExecuteOperation(TestWorkspace workspace, Action<TestWorkspace> operation)
```

Comment:
😟 This pair of overloads seems like it could be confusing. Do we need this one that has a synchronous callback?

Reply:
> Do we need this one that has a synchronous callback?

Yes.  Many/most of the callers in this file use the synchronous version, making workspace changes with the real synchronous methods it provides.  Without having this overload, we'd need all of those to update to be fake-async, or to return Task.Completed task.

NOte: all this code goes away anyways when we remove solution crawler.  This is test code and i don't see much reason to fret over this.